### PR TITLE
Bump CAPI to v1.5.1

### PR DESCRIPTION
**Pull request by haijianyang**
Docs:
- https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.5.1
- https://main.cluster-api.sigs.k8s.io/developer/providers/migrations/v1.4-to-v1.5.html

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -37,88 +47,147 @@ import (
...
-		"health-addr",
-		":9440",
-		"The address the health endpoint binds to.",
-	)
-	flag.IntVar(
-		&config.MaxConcurrentVMCreations,
-		"max-concurrent-vm-creations",
-		config.MaxConcurrentVMCreations,
+
+	fs.IntVar(&elfClusterConcurrency, "elfcluster-concurrency", 10,
```
Comment (jessehu) [inline main.go:87]: elfcluster-concurrency => max-elfcluster-concurrent-reconciles

File: main.go [hunk 2]
```diff
@@ -37,88 +47,147 @@ import (
...
+		"Number of ELF clusters to process simultaneously")
+
+	fs.IntVar(&elfMachineConcurrency, "elfmachine-concurrency", 10,
```
Comment (jessehu) [inline main.go:90]: elfmachine-concurrency => max-elfmachine-concurrent-reconciles

File: config/manager/manager.yaml [hunk 3]
```diff
@@ -18,7 +18,6 @@ spec:
       containers:
       - args:
         - --leader-elect
-        - --logtostderr
```
[the remark this answers is not in the document]
Comment (haijianyang) [reply config/manager/manager.yaml:21]: https://cluster-api.sigs.k8s.io/developer/providers/migrations/v1.3-to-v1.4.html?highlight=logtostderr#other

**Review by jessehu — APPROVED** [review]
nice

**Pull request merged**